### PR TITLE
Feature/trustly charge

### PR DESCRIPTION
**Pull request by EtimoOscarNylander**
Functionality missing:
- ~~Read model~~
- ~~Exposing the read model as api~~
- ~~Saving pending notifications as events~~

Tests missing:

- Sad path for notifications
- Happy path for all kinds of notificiations
- Possibly some other tests

### CODE REVIEW

**Comment by EtimoOscarNylander** [comment]
I'm going to merge this to test it on the dev cluster and to start working on the next trustly-related feature - we can solve all issues that come up from the CR in another branch.

File: payment-service/src/test/java/com/hedvig/paymentservice/trustly/testHelpers/TestData.java [hunk 1]
```diff
@@ -1,17 +1,98 @@
...
-                TRIGGER_ID);
+                TOLVAN_FIRST_NAME,
+                TOLVANSSON_LAST_NAME,
+                TOLVANSSON_SSN,
+                TOLVAN_EMAIL,
+                MEMBER_ID,
+                BOT_SERVICE_TRIGGER_ID);
+    }
+
+    public static TrustlyAccountCreatedEvent createTrustlyAccountCreatedEvent() {
```
Comment (johantjellden) [inline payment-service/src/test/java/com/hedvig/paymentservice/trustly/testHelpers/TestData.java]: rename to makeTrustlyAccountCreatedEvent

File: payment-service/src/test/java/com/hedvig/paymentservice/trustly/testHelpers/TestData.java [hunk 2]
```diff
@@ -1,17 +1,98 @@
...
+            TRUSTLY_ACCOUNT_DESCRIPTOR,
+            TRUSTLY_ACCOUNT_DIRECTDEBIT_TRUE,
+            TRUSTLY_ACCOUNT_LAST_DIGITS,
+            TOLVAN_FIRST_NAME,
+            TOLVANSSON_SSN,
+            TOLVANSSON_ZIP
+        );
+    }
+
+    public static Notification createTrustlyCreditNotificationRequest() {
```
Comment (johantjellden) [inline payment-service/src/test/java/com/hedvig/paymentservice/trustly/testHelpers/TestData.java]: Rename to makeTrustlyCreditNotificationRequest

File: payment-service/src/test/java/com/hedvig/paymentservice/web/internal/MemberControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                TRUSTLY_ACCOUNT_CLEARING_HOUSE,
+                TRUSTLY_ACCOUNT_DESCRIPTOR,
+                TRUSTLY_ACCOUNT_DIRECTDEBIT_TRUE,
+                TRUSTLY_ACCOUNT_LAST_DIGITS,
+                TOLVAN_FIRST_NAME + " " + TOLVANSSON_LAST_NAME,
+                TOLVANSSON_SSN,
+                TOLVANSSON_ZIP
+                ));
+
+        mockTrustlyApiResponse(true);
```
Comment (johantjellden) [inline payment-service/src/test/java/com/hedvig/paymentservice/web/internal/MemberControllerTest.java]: Create a named constant from boolean

File: payment-service/src/test/java/com/hedvig/paymentservice/web/internal/MemberControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+                post(
+                    String.format(
+                        "/_/members/%s/charge",
+                        MEMBER_ID
+                    )
+                )
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(chargeRequest))
+            )
+            .andExpect(status().is(202));
```
Comment (johantjellden) [inline payment-service/src/test/java/com/hedvig/paymentservice/web/internal/MemberControllerTest.java]: Use HttpStatusCode* enum

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/TrustlyOrder.java [hunk 5]
```diff
@@ -42,73 +42,92 @@ public TrustlyOrder(CreateOrderCommand cmd) {
...
 
-        if(notifications.contains(data.getNotificationId())) {
+    @CommandHandler
+    public void cmd(AccountNotificationReceivedCommand cmd) {
+
+        if(handledNotifications.contains(cmd.getNotificationId())) {
             return;
         }
 
+        apply(new AccountNotificationReceivedEvent(
```
[the remark this answers is not in the document]
Comment (EtimoOscarNylander) [reply payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/TrustlyOrder.java]: @johantjellden It looks like we do not get an endUserId for the 'Account'-type notification from Trustly :disappointed:

**Pull request merged**